dkpro-core-arktools-gpl/src/main/java/de/tudarmstadt/ukp/dkpro/core/arktools/ArktweetPosTaggerTrainer.java

```diff
@@ -0,0 +1,97 @@
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+package de.tudarmstadt.ukp.dkpro.core.arktools;
+
+import static org.apache.uima.fit.util.JCasUtil.indexCovered;
+import static org.apache.uima.fit.util.JCasUtil.select;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.uima.analysis_engine.AnalysisEngineProcessException;
+import org.apache.uima.fit.component.JCasConsumer_ImplBase;
+import org.apache.uima.fit.descriptor.ConfigurationParameter;
+import org.apache.uima.fit.descriptor.MimeTypeCapability;
+import org.apache.uima.fit.descriptor.ResourceMetaData;
+import org.apache.uima.jcas.JCas;
+
+import cmu.arktweetnlp.Train;
+import de.tudarmstadt.ukp.dkpro.core.api.parameter.ComponentParameters;
+import de.tudarmstadt.ukp.dkpro.core.api.parameter.MimeTypes;
+import de.tudarmstadt.ukp.dkpro.core.api.segmentation.type.Sentence;
+import de.tudarmstadt.ukp.dkpro.core.api.segmentation.type.Token;
+
+@MimeTypeCapability(MimeTypes.APPLICATION_X_ARKTWEET_TAGGER)
+@ResourceMetaData(name = "ArkTweet POS-Tagger Trainer")
+public class ArktweetPosTaggerTrainer extends JCasConsumer_ImplBase {
+
+    public static final String PARAM_TARGET_LOCATION = ComponentParameters.PARAM_TARGET_LOCATION;
+    @ConfigurationParameter(name = PARAM_TARGET_LOCATION, mandatory = true)
+    private File targetLocation;
+
+    private File tempData;
+    private PrintWriter out;
+
+    @Override
+    public void process(JCas jCas) throws AnalysisEngineProcessException {
+        if (tempData == null) {
+            try {
+                tempData = File.createTempFile("dkpro-stanford-pos-trainer", ".tsv");
+                out = new PrintWriter(new OutputStreamWriter(new FileOutputStream(tempData),
+                        StandardCharsets.UTF_8));
+            }
+            catch (IOException e) {
+                throw new AnalysisEngineProcessException(e);
+            }
+        }
+
+
+        Map<Sentence, Collection<Token>> index = indexCovered(jCas, Sentence.class, Token.class);
+        for (Sentence sentence : select(jCas, Sentence.class)) {
+            Collection<Token> tokens = index.get(sentence);
+            for (Token token : tokens) {
+                out.printf("%s\t%s%n", token.getText(), token.getPos().getPosValue());
+            }
+            out.println();
+        }
+    }
+
+    @Override
+    public void collectionProcessComplete() throws AnalysisEngineProcessException {
+        if (out != null) {
+            IOUtils.closeQuietly(out);
+        }
+
+        try {
+            Train.main(new String[] {
+                    tempData.toString(),
+                    targetLocation.toString()
+            });
+        } catch (IOException e) {
```

Comment:
The temporary file should be deleted at the end of the processing. Best overwrite the `destroy()` method and clean the temporary file up there because `collectionProcessComplete()` may not be called in case there is an error during processing. However, `destroy()` should always be called to let the component clean itself up.

Reply:
Done.